### PR TITLE
Redesign Navigation: Mobile Implementation

### PR DESCRIPTION
**Pull request by kylorhall**
Building off of #249 which is built off of #248.

This PR will ideally will really just encapsulate all feedback merged straight into #248, but I felt this isolation of work was helpful in review phase.

Videos so far:
Desktop: https://www.loom.com/share/cce25ed2d0494da7ad093412190d0c54
Mobile: https://www.loom.com/share/7b8f4d425c6d4d2a905a608523a541ae

I think this is ready for testing now, so long as 

Still needs:
 - [x] A Ruby code review.
 - [ ] A whole slew of test coverage!
 - [ ] A bit of content and input (moving things around or hiding some links).
 - [ ] A lot of QA, Accessibility Testing, Browser Testing, etc.  I haven't done much in that sense.
 - [ ] A bit more in terms of tweaking some design.

### CODE REVIEW

File: source/js/src/navigation.js.es6 [hunk 1]
```diff
@@ -1,3 +1,31 @@
+function handleKeyboardAction ({ event, onOpen, onEscape }) {
```
Comment (irwalker) [inline source/js/src/navigation.js.es6:1]: why?

File: source/partials/header/_row.html.erb [hunk 2]
```diff
@@ -4,16 +4,10 @@
...
-      <% if column[:label] %>
-        <div class="menu__title" role="separator"><%= column[:label] %></div>
-      <% end %>
-
-      <% column[:links].each do |link| %>
-        <%= partial "partials/header/link", locals: link %>
-      <% end %>
-
-    </div> <%# closes "menu__split-#" %>
+    <%= partial "partials/header/column", locals: {
```
Comment (irwalker) [inline source/partials/header/_row.html.erb:7]: 👍

File: source/partials/header/_navigation.html.erb [hunk 3]
```diff
@@ -21,20 +21,17 @@
               label: 'Features',
               links: [
                 {
-                  compact: true,
```
Comment (irwalker) [inline source/partials/header/_navigation.html.erb:24]: Let's do the "move this into a helper" bit

**Review by irwalker — COMMENTED** [review]
Looks good actually, I think one of the front-end team should really review the JS and CSS, I can't do a proper review of that. But let's add tests etc before this gets merged

File: source/js/src/navigation.js.es6 (hunk 1 above)
Comment (kylorhall) [reply source/js/src/navigation.js.es6:1]: Just reduces a ton of copy and paste—we use 80% of this function in 5 places.

It is basically for accessibility:
 - I click `space` or `enter` and I want to simulate a "click" and do that something.
 - I click `escape` and I want to "exit".

Comment (kylorhall) [reply source/js/src/navigation.js.es6:1]: Will definitely request a review on the JS side!

**Comment by kylorhall** [comment]
Into #248 now.

**Pull request merged**